### ISSUE TITLE
Rule to enforce default only import from JSON module

### ISSUE THREAD

**Issue by alspdx** [import-js/eslint-plugin-import#2359]
I'm trying to find a way to enforce default only imports from files with a `.json` extension, I wasn't able to find a rule that fits. 

The TC39 proposal for [JSON Modules](https://github.com/tc39/proposal-json-modules) is at stage 3, and Webpack 5 emits a warning when using a non-default import from a JSON file:
```js
// config.json
{
  "path": "/somewhere"
}

// some-file.js
import { path } from './config.json'

// Webpack compiler warning:
// ⚠️ Should not import the named export 'path' (imported as 'path') from default-exporting module (only default export is available soon)
```

From the [Webpack 5 release announcement](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#json-modules) (Oct 10, 2020):
> JSON modules now align with the proposal and emit a warning when a non-default export is used. JSON modules no longer have named exports when importing from a strict ECMAScript module.
MIGRATION: Use the default export.

The rule might look something like:
```yaml
rules:
  no-named-import: [2, { json: true }]
  # or
  only-default-import: [2, { json: true }]
```
```js
import { path } from './config.json' ❌ 

import config from './config.json' ✅ 
```
Absolutely open to suggestions if there's anything I should try, thanks in advance!

**Comment by ljharb** [comment]
I’d expect the “named” rule to do this.

**Comment by alspdx** [comment]
Thanks — the documentation on the rule doesn't mention anything about restricting it to a specific file type, is that configurable?

**Comment by ljharb** [comment]
What i mean is, zero json files have a named export, so I’d expect the rule to warn on them with no configuration required. Have you tried? (the Airbnb config enables this for you)

**Comment by alspdx** [comment]
Doesn't change anything for me. I have the Airbnb Typescript config enabled, and I just manually configured the rule — tried with named imports in TS and JS files.

**Comment by ljharb** [comment]
There is no actual Airbnb typescript config - the one you’re using is unofficial - but I’m sure that doesn’t change things here.

If the named rule doesn’t cover it, then that’s surprising. Is your TS config set to resolveJsonModule: true?

**Comment by alspdx** [comment]
Right, yes, it's the [eslint-config-airbnb-typescript](https://github.com/iamturns/eslint-config-airbnb-typescript) plugin. 

Yes, TS config is set to resolve JSON modules. As far as I was aware that was required for TS to import JSON. I realize that it does also generate a static type for the JSON module as well, would that affect how the module is resolved?

**Comment by ljharb** [comment]
Yes, it would. the TS resolver for this plugin uses the type.

If the type of a JSON module doesn't indicate that it only has a default export, then that type is wrong. If it's TS generating that type, I'd file a bug on TS.

**Comment by alspdx** [comment]
Ok, thank you for taking the time to help sort this out!
